test/unitary/IQueryable.test.ts

```diff
@@ -1286,6 +1306,61 @@ export namespace IQueryableUnitTest
         });
     }
 
+    function takeWhile(instancer: Instancer): void
+    {
+        it("Empty if empty (true)", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(base.takeWhile(e => true).toArray(), []);
+        });
+
+        it("Empty if empty (false)", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(base.takeWhile(e => false).toArray(), []);
+        });
+
+        it("Empty if empty (true) (iterator)", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(new Enumerable(base.takeWhile(e => true)).toArray(), []);
+        });
+
+        it("Empty if empty (false) (iterator)", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(new Enumerable(base.takeWhile(e => false)).toArray(), []);
+        });
+
+        it("Value is correct (returns elements)", () =>
+        {
+            const base = instancer([39, 40, 21, 66, 20]);
+            const takeWhile = base.takeWhile(e => e >= 39);
```

Comment:
shadowing function name, give it another name

Reply:
on all of them

Reply:
i'm not sure why it reports shadowing since it's identical code to skipWhile, where, etc... hmmm

Reply:
hmmmm that's strange, yep
nevermind, the PR is fine